### PR TITLE
Update sdk version for testing from 5 to 6 (6.0.3xx)

### PR DESCRIPTION
**Pull request by heng-liu**
## Bug

Fixes:https://github.com/NuGet/Client.Engineering/issues/1271

Regression? Last working version:

## Description
Update the sdk version for testing from 5 to 6 and fix a bunch of "consequences".
For better understanding of those changes and faster review, here are some explanations:
1. **build/common.ps1** : simplify the logic based on some consideration in this [comment](https://github.com/NuGet/NuGet.Client/pull/4400#discussion_r783568435)
2. **eng/pipelines/templates/Build_and_UnitTest.yml**: install nuget tool 5.x or else the default is 4.x then the succeeding nuget sign task would fail as there is no verify command in nuget 4.x
3. move from VSEngSS-MicroBuild2019-1ES to VSEngSS-MicroBuild2022-1ES/install 6.x .NET SDK, as we need to change TFM of dotnet.integration.tests from net5 to net6, so we need to build this test project with .net 6 SDK. 
4. add task "Set MSBuildEnableWorkloadResolver to false" as a workaround or else test would fail for `error MSB4236: The SDK 'Microsoft.NET.SDK.WorkloadAutoImportPropsLocator' specified could not be found.`
5. Change **setup/Microsoft.VisualStudio.NuGet.BuildTools.vsmanproj** and lots of net40/net45/net451 in the tests as VS2022 doesn't support net45 or before anymore. Since we are moving to VSEngSS-MicroBuild2022-1ES pool, we need to get rid of net45. See [Visual Studio 2022 Support for .NET Development](https://docs.microsoft.com/en-us/visualstudio/releases/2022/compatibility#-visual-studio-2022-support-for-net-development) for more details.
6. Change **test/NuGet.Core.FuncTests/Dotnet.Integration.Test/MsbuildIntegrationTestFixture.cs** to adapt  the change when the [default C# compiler changed to a new version 10 in .NET 6](https://docs.microsoft.com/en-us/dotnet/csharp/language-reference/configure-language-version).
7. remove a test template `reactredux` and change `solution  `to `slu` as dotnet template team said this is by design in https://github.com/dotnet/templating/issues/4825
8. change net5.0 to net6.0 in dotnet.integration.tests as we only install 6.0 under cli folder, tests with net5.0 would fail. 
9. refactor **test/TestUtilities/Test.Utility/TestDotnetCLiUtility.cs** so that it could handle the case when there are multiple testing SDK versions.
10. change the sequence of installing .NET SDK/runtimes as a workaround: install .NET 6 SDK first, then install .NET 5 runtimes, finally install .NETCoreapp3.1 runtimes. Thanks for @kartheekp-ms finding this workaround to avoid macos tests failure.
11. Add MSTest.TestAdapter and MSTest.TestFramework into NuGet.Config file, package source mapping section. So it will be restored from dotnet-public source(package source mapping is enabled). Otherwise, projects NuGet.OptProf and NuGet.Tests.Apex will fail the restore.

## PR Checklist

- [x] PR has a meaningful title
- [x] PR has a linked issue.
- [x] Described changes

- **Tests**
  - [ ] Automated tests added
  - **OR**
  
  - [ ] Test exception
  - **OR**
  - [x] N/A 

- **Documentation**
  
  - [ ] Documentation PR or issue filled
  - **OR**
  - [x] N/A

### CODE REVIEW

**Comment by heng-liu** [comment]
//cc @dtivel as this PR would unblock https://github.com/NuGet/NuGet.Client/pull/4585

File: eng/pipelines/templates/Build_and_UnitTest.yml
Comment (nkolev92) [inline eng/pipelines/templates/Build_and_UnitTest.yml:222]: Which component was installing 4.x?

Can we do 6.x instead?

File: eng/pipelines/templates/Functional_Tests_On_Windows.yml
Comment (nkolev92) [inline eng/pipelines/templates/Functional_Tests_On_Windows.yml]: Is this something we'd want to unset? 
Would having this setting cause any problems running these tests locally?

File: scripts/funcTests/runFuncTests.sh
Comment (nkolev92) [inline scripts/funcTests/runFuncTests.sh]: recommended?

File: scripts/funcTests/runFuncTests.sh
Comment (nkolev92) [inline scripts/funcTests/runFuncTests.sh:96]: I imagine this matches the tfms our tests are targetting?

File: setup/Microsoft.VisualStudio.NuGet.BuildTools.vsmanproj
Comment (nkolev92) [inline setup/Microsoft.VisualStudio.NuGet.BuildTools.vsmanproj]: What was the previous version here? 
Asking because it's only an addition

File: test/NuGet.Clients.Tests/NuGet.CommandLine.Test/NuGetPackCommandTest.cs
Comment (nkolev92) [inline test/NuGet.Clients.Tests/NuGet.CommandLine.Test/NuGetPackCommandTest.cs]: how did this change to net48?
I'm not sure which particular change caused it?

**Review by nkolev92 — COMMENTED** [review]
Nicely done @heng-liu! 

Most of my comments are asking for some clarifications to make sure I understand the changes.

File: build/config.props
Comment (dtivel) [inline build/config.props]: Why not `6.0`?  Wouldn't that pick up 6.0.4xx by default?

For `7.0`, I hope we can just put `7.0` and get the latest branch by default.

File: build/common.ps1
Comment (dtivel) [inline build/common.ps1:210]: Does `signed` imply *daily* signed builds?

File: eng/pipelines/templates/Build_and_UnitTest.yml
Comment (dtivel) [inline eng/pipelines/templates/Build_and_UnitTest.yml]: Why not 6.x?

File: build/config.props
Comment (heng-liu) [reply build/config.props]: I just tested, `6.0` is the same with `6.0.3xx`, it will install 6.0.3, not 6.0.4.
```
.\dotnet-install.ps1 -Channel 6.0 -Quality signed -InstallDir .\cli -Version latest -NoPath
dotnet-install: Note that the intended use of this script is for Continuous Integration (CI) scenarios, where:
dotnet-install: - The SDK needs to be installed without user interaction and without admin rights.
dotnet-install: - The SDK installation doesn't need to persist across multiple CI runs.
dotnet-install: To set up a development environment or to run apps, use installers rather than this script. Visit https://dotnet.microsoft.com/download to get the installer.

dotnet-install: Using alternate version 6.0.300 found in https://dotnetstage.blob.core.windows.net/6-0-300-rtm-22213-34-public/Sdk/6.0.300-rtm.22213.34/sdk-productVersion.txt
dotnet-install: Extracting the archive.
dotnet-install: Binaries of dotnet can be found in C:\repos\NuGet.Client\cli\cli\
dotnet-install: Note that the script does not resolve dependencies during installation.
dotnet-install: To check the list of dependencies, go to https://docs.microsoft.com/dotnet/core/install/windows#dependencies
dotnet-install: Installation finished
```

File: eng/pipelines/templates/Build_and_UnitTest.yml
Comment (heng-liu) [reply eng/pipelines/templates/Build_and_UnitTest.yml:222]: The agent pool VSEngSS-MicroBuild2022-1ES will use NuGet tool with 4.x by default.
Previously I checked our official build, it uses 5.8.0 NuGet tool for verifying: https://devdiv.visualstudio.com/DevDiv/_build/results?buildId=6297556&view=logs&j=d80b2849-30d4-52dd-74cd-d6536ba98fe0&t=df8f2bf5-5d3a-5e1a-c7c8-4bd106082de1&l=14
Sure, I will change it to 6.x.

File: eng/pipelines/templates/Build_and_UnitTest.yml
Comment (heng-liu) [reply eng/pipelines/templates/Build_and_UnitTest.yml]: Previously I checked our official build, it uses 5.8.0 NuGet tool for verifying: https://devdiv.visualstudio.com/DevDiv/_build/results?buildId=6297556&view=logs&j=d80b2849-30d4-52dd-74cd-d6536ba98fe0&t=df8f2bf5-5d3a-5e1a-c7c8-4bd106082de1&l=14
Sure, I will change it to 6.x.

File: test/NuGet.Clients.Tests/NuGet.CommandLine.Test/NuGetPackCommandTest.cs
Comment (heng-liu) [reply test/NuGet.Clients.Tests/NuGet.CommandLine.Test/NuGetPackCommandTest.cs]: ~~Since we are moving to VSEngSS-MicroBuild2022-1ES pool, we need to get rid of net45. Please see [Visual Studio 2022 Support for .NET Development](https://docs.microsoft.com/en-us/visualstudio/releases/2022/compatibility#-visual-studio-2022-support-for-net-development) for more details.~~

File: scripts/funcTests/runFuncTests.sh
Comment (heng-liu) [reply scripts/funcTests/runFuncTests.sh:96]: That right! We have test projects targeting netcoreapp3.1 and/or net5.0, and dotnet.integration.tests targeting net6.0.
So we install net3.1 and net5.0 runtimes, and net6.0 SDK, then invoke the dotnet test and passes all the test assemblies together.

File: setup/Microsoft.VisualStudio.NuGet.BuildTools.vsmanproj
Comment (heng-liu) [reply setup/Microsoft.VisualStudio.NuGet.BuildTools.vsmanproj]: There is no specific version there and it's net4.0.
So it's not able to be built on VS2022, as VS2022 doesn't support net45(or earlier full framework) anymore.
Please see [Visual Studio 2022 Support for .NET Development](https://docs.microsoft.com/en-us/visualstudio/releases/2022/compatibility#-visual-studio-2022-support-for-net-development) for more details.

Comment (nkolev92) [reply setup/Microsoft.VisualStudio.NuGet.BuildTools.vsmanproj]: Can we do net472 instead as that's what our projects actually target?

File: test/NuGet.Clients.Tests/NuGet.CommandLine.Test/NuGetPackCommandTest.cs
Comment (nkolev92) [reply test/NuGet.Clients.Tests/NuGet.CommandLine.Test/NuGetPackCommandTest.cs]: My question is more about where's the code that made this go from net40 to net48. I had a hard time finding it

File: scripts/funcTests/runFuncTests.sh
Comment (heng-liu) [reply scripts/funcTests/runFuncTests.sh]: Changed the comment into `# Install .NET 5 runtimes and .NETCoreapp3.1 runtimes`

File: setup/Microsoft.VisualStudio.NuGet.BuildTools.vsmanproj
Comment (heng-liu) [reply setup/Microsoft.VisualStudio.NuGet.BuildTools.vsmanproj]: Sure, I'll change it into net472. Thanks!

File: eng/pipelines/templates/Build_and_UnitTest.yml
Comment (heng-liu) [reply eng/pipelines/templates/Build_and_UnitTest.yml:222]: Changed to 6.x.

File: eng/pipelines/templates/Build_and_UnitTest.yml
Comment (heng-liu) [reply eng/pipelines/templates/Build_and_UnitTest.yml]: Changed to 6.x.

File: build/common.ps1
Comment (heng-liu) [reply build/common.ps1:210]: Good question!
Just asked .NET team, signed doesn't imply a daily signed build.
Michelle's reply:
It is not a daily build, no. The intent is that it is a pre-release of the currently in-flight build that is currently being considered for release. Validated are those builds that have been chosen as the release build, but before actual release day.

File: scripts/funcTests/runFuncTests.sh
Comment (nkolev92) [reply scripts/funcTests/runFuncTests.sh:96]: Thanks, resolving this.

File: test/NuGet.Clients.Tests/NuGet.CommandLine.Test/NuGetPackCommandTest.cs
Comment (heng-liu) [reply test/NuGet.Clients.Tests/NuGet.CommandLine.Test/NuGetPackCommandTest.cs]: You're right! Those `net40` won't failed the tests as there is no need to create projects.
Revert changes in this file. Thanks!

File: eng/pipelines/templates/Functional_Tests_On_Windows.yml
Comment (heng-liu) [reply eng/pipelines/templates/Functional_Tests_On_Windows.yml]: This env var is not set locally and the test passes.
If I set `MSBuildEnableWorkloadResolver` to true locally, the test (e.g. `AddPkg_ConditionalAddWithUserInputFramework_Success`) fails for the same reason:
```
Message: 
System.AggregateException : One or more errors occurred.
---- Microsoft.Build.Exceptions.InvalidProjectFileException : The SDK 'Microsoft.NET.SDK.WorkloadAutoImportPropsLocator' specified could not be found.  C:\Program Files\dotnet\sdk\5.0.409\Sdks\Microsoft.NET.Sdk\targets\Microsoft.NET.Sdk.ImportWorkloads.props

  Stack Trace: 
Task.ThrowIfExceptional(Boolean includeTaskCanceledExceptions)
Task`1.GetResultCore(Boolean waitCompletionNotification)
Task`1.get_Result()
<AddPkg_ConditionalAddWithUserInputFramework_Success>d__17.MoveNext() line 673
--- End of stack trace from previous location where exception was thrown ---
TaskAwaiter.ThrowForNonSuccess(Task task)
TaskAwaiter.HandleNonSuccessAndDebuggerNotification(Task task)
--- End of stack trace from previous location where exception was thrown ---
TaskAwaiter.ThrowForNonSuccess(Task task)
TaskAwaiter.HandleNonSuccessAndDebuggerNotification(Task task)
<17 more frames...>
Evaluator`4.Evaluate(IEvaluatorData`4 data, ProjectRootElement root, ProjectLoadSettings loadSettings, Int32 maxNodeCount, PropertyDictionary`1 environmentProperties, ILoggingService loggingService, IItemFactory`2 itemFactory, IToolsetProvider toolsetProvider, ProjectRootElementCacheBase projectRootElementCache, BuildEventContext buildEventContext, ISdkResolverService sdkResolverService, Int32 submissionId, EvaluationContext evaluationContext, Boolean interactive)
ProjectImpl.Reevaluate(ILoggingService loggingServiceForEvaluation, ProjectLoadSettings loadSettings, EvaluationContext evaluationContext)
ProjectImpl.ReevaluateIfNecessary(ILoggingService loggingServiceForEvaluation, ProjectLoadSettings loadSettings, EvaluationContext evaluationContext)
ProjectImpl.ReevaluateIfNecessary(EvaluationContext evaluationContext)
ProjectImpl.Initialize(IDictionary`2 globalProperties, String toolsVersion, String subToolsetVersion, ProjectLoadSettings loadSettings, EvaluationContext evaluationContext)
Project.ctor(ProjectRootElement xml, IDictionary`2 globalProperties, String toolsVersion, String subToolsetVersion, ProjectCollection projectCollection, ProjectLoadSettings loadSettings, EvaluationContext evaluationContext)
Project.ctor(ProjectRootElement xml, IDictionary`2 globalProperties, String toolsVersion)
MSBuildAPIUtility.GetProject(String projectCSProjPath, IDictionary`2 globalProperties) line 71
MSBuildAPIUtility.AddPackageReferencePerTFM(String projectPath, LibraryDependency libraryDependency, IEnumerable`1 frameworks) line 148
<ExecuteCommand>d__0.MoveNext() line 240
```
But this env var is not set when running on CI eithor. Not sure if the default value changed.

File: NuGet.Config
Comment (zivkan) [inline NuGet.Config:25]: I can't find where these are being used. Can it be removed?

File: build/config.props
Comment (zivkan) [inline build/config.props:46]: Should be put a comment next to the major/minor version numbers near the top, since every major/minor version maps to a different .NET SDK version?

File: eng/pipelines/templates/Build_and_UnitTest.yml
Comment (zivkan) [inline eng/pipelines/templates/Build_and_UnitTest.yml:225]: [NuGetToolInstaller](https://docs.microsoft.com/en-us/azure/devops/pipelines/tasks/tool/nuget?view=azure-devops) has a `checkLatest`. Should we use that, so we don't need to maintain this version as NuGet bumps major version numbers?

File: eng/pipelines/templates/Functional_Tests_On_Windows.yml
Comment (zivkan) [reply eng/pipelines/templates/Functional_Tests_On_Windows.yml]: workloads are net to .NET 6, so they didn't exist in the .NET 5 SDK.  At this time, I believe it's only used by maui and xamarin, which we don't have tests for at this time, hence it doesn't break us (yet?).

File: eng/pipelines/templates/Functional_Tests_On_Windows.yml
Comment (zivkan) [inline eng/pipelines/templates/Functional_Tests_On_Windows.yml]: Environment variables can be set more easily several ways:

* `variables` parameter on a job:

This sets the environment variable for all tasks in the job.

```yaml
jobs:
- job: jobName
  variables:
    MSBuildEnableWorkloadResolver: false
  steps:
```

* `env:` parameter on tasks

```yaml
- task: PowerShell@2
  filePath: path/to/script.ps1
  arguments: "-Arg value"
  env:
    MSBuildEnableWorkloadRes: false
```

* `variables:` at the pipeline definition

This makes the variable set for all stages, all jobs, all tasks. We already use it for something else: https://github.com/NuGet/NuGet.Client/blob/495a88382f5b357e62e83667a5f05192ee33e226/eng/pipelines/templates/pipeline.yml#L45-L46

File: eng/pipelines/templates/CrossFramework_Tests_On_Windows.yml
Comment (zivkan) [inline eng/pipelines/templates/CrossFramework_Tests_On_Windows.yml:7]: Any reason not to let the task use its own default installation path?

File: setup/Microsoft.VisualStudio.NuGet.BuildTools.vsmanproj
Comment (zivkan) [inline setup/Microsoft.VisualStudio.NuGet.BuildTools.vsmanproj:13]: Out of curiosity, do you remember why this is needed?  This file builds NuGet's "package" for the VS installer. I wouldn't have expected it to need changes when changing the SDK version for `dotnet.integration.tests`. I guess moving the build to the VS2022 agent pool might have caused something to change. I'm just surprised.

File: eng/pipelines/templates/Functional_Tests_On_Windows.yml
Comment (heng-liu) [reply eng/pipelines/templates/Functional_Tests_On_Windows.yml]: I tried to remove this workaround. There are multiple tests in NuGet.XPlat.FuncTest.XPlatAddPkgTests failed.
The build is: https://devdiv.visualstudio.com/DevDiv/_build/results?buildId=6304121&view=results

File: eng/pipelines/templates/Build_and_UnitTest.yml
Comment (heng-liu) [reply eng/pipelines/templates/Build_and_UnitTest.yml:225]: Thanks! I noticed that option too. Not sure if we should do that as it says `Enabling this option could cause unexpected build breaks when a new version of NuGet is released`.

File: eng/pipelines/templates/CrossFramework_Tests_On_Windows.yml
Comment (heng-liu) [reply eng/pipelines/templates/CrossFramework_Tests_On_Windows.yml:7]: According to [doc](https://docs.microsoft.com/en-us/azure/devops/pipelines/tasks/tool/dotnet-core-tool-installer?view=azure-devops#task-inputs):
```
$(Agent.ToolsDirectory): This makes the version to be cached on the agent since this directory is not cleanup up across pipelines. All pipelines running on the agent, would have access to the versions installed previously using the agent.
$(Agent.TempDirectory): This can ensure that a pipeline doesn't use any cached version of .NET core since this folder is cleaned up after each pipeline.
Default value: $(Agent.ToolsDirectory)/dotnet
```
The default value `$(Agent.ToolsDirectory)/dotnet` will not be cleanup across pipelines. But `$(Agent.TempDirectory)` will be cleaned up after each pipeline. I think we do not want to pollute the agent machine to affect others' builds.

File: setup/Microsoft.VisualStudio.NuGet.BuildTools.vsmanproj
Comment (heng-liu) [reply setup/Microsoft.VisualStudio.NuGet.BuildTools.vsmanproj:13]: The setup\Microsoft.VisualStudio.NuGet.BuildTools.vsmanproj is targeting net4.0, so it is not able to be built on VS2022.
According to [this doc](https://docs.microsoft.com/en-us/dotnet/framework/install/guide-for-developers), Visual Studio 2022 and later versions can't build apps that target .NET Framework 4.0 through .NET Framework 4.5.1.

File: NuGet.Config
Comment (heng-liu) [reply NuGet.Config:25]: Seems no. Pls check this [build ](https://devdiv.visualstudio.com/DevDiv/_build/results?buildId=6305254&view=results):
```
##[error]test\NuGet.Tests.Apex\NuGet.OptProf\NuGet.OptProf.csproj(0,0): Error NU1100: Unable to resolve 'MSTest.TestAdapter (>= 1.2.0)' for '.NETFramework,Version=v4.7.2'. PackageSourceMapping is enabled, the following source(s) were not considered: dotnet-eng, dotnet-public, myget-legacy@Local, nuget-build, vside_vs-impl, vside_vssdk.
D:\a\_work\1\s\test\NuGet.Tests.Apex\NuGet.OptProf\NuGet.OptProf.csproj : error NU1100: Unable to resolve 'MSTest.TestAdapter (>= 1.2.0)' for '.NETFramework,Version=v4.7.2'. PackageSourceMapping is enabled, the following source(s) were not considered: dotnet-eng, dotnet-public, myget-legacy@Local, nuget-build, vside_vs-impl, vside_vssdk. [D:\a\_work\1\s\build\restorehelper.targets]
##[error]test\NuGet.Tests.Apex\NuGet.OptProf\NuGet.OptProf.csproj(0,0): Error NU1100: Unable to resolve 'MSTest.TestFramework (>= 1.2.0)' for '.NETFramework,Version=v4.7.2'. PackageSourceMapping is enabled, the following source(s) were not considered: dotnet-eng, dotnet-public, myget-legacy@Local, nuget-build, vside_vs-impl, vside_vssdk.
D:\a\_work\1\s\test\NuGet.Tests.Apex\NuGet.OptProf\NuGet.OptProf.csproj : error NU1100: Unable to resolve 'MSTest.TestFramework (>= 1.2.0)' for '.NETFramework,Version=v4.7.2'. PackageSourceMapping is enabled, the following source(s) were not considered: dotnet-eng, dotnet-public, myget-legacy@Local, nuget-build, vside_vs-impl, vside_vssdk. [D:\a\_work\1\s\build\restorehelper.targets]
##[error]test\NuGet.Tests.Apex\NuGet.Tests.Apex\NuGet.Tests.Apex.csproj(0,0): Error NU1100: Unable to resolve 'MSTest.TestAdapter (>= 1.2.0)' for '.NETFramework,Version=v4.7.2'. PackageSourceMapping is enabled, the following source(s) were not considered: dotnet-eng, dotnet-public, myget-legacy@Local, nuget-build, vside_vs-impl, vside_vssdk.
D:\a\_work\1\s\test\NuGet.Tests.Apex\NuGet.Tests.Apex\NuGet.Tests.Apex.csproj : error NU1100: Unable to resolve 'MSTest.TestAdapter (>= 1.2.0)' for '.NETFramework,Version=v4.7.2'. PackageSourceMapping is enabled, the following source(s) were not considered: dotnet-eng, dotnet-public, myget-legacy@Local, nuget-build, vside_vs-impl, vside_vssdk. [D:\a\_work\1\s\build\restorehelper.targets]
##[error]test\NuGet.Tests.Apex\NuGet.Tests.Apex\NuGet.Tests.Apex.csproj(0,0): Error NU1100: Unable to resolve 'MSTest.TestFramework (>= 1.2.0)' for '.NETFramework,Version=v4.7.2'. PackageSourceMapping is enabled, the following source(s) were not considered: dotnet-eng, dotnet-public, myget-legacy@Local, nuget-build, vside_vs-impl, vside_vssdk.
D:\a\_work\1\s\test\NuGet.Tests.Apex\NuGet.Tests.Apex\NuGet.Tests.Apex.csproj : error NU1100: Unable to resolve 'MSTest.TestFramework (>= 1.2.0)' for '.NETFramework,Version=v4.7.2'. PackageSourceMapping is enabled, the following source(s) were not considered: dotnet-eng, dotnet-public, myget-legacy@Local, nuget-build, vside_vs-impl, vside_vssdk. [D:\a\_work\1\s\build\restorehelper.targets]
##[error]test\NuGet.Tests.Apex\NuGet.Tests.Apex\NuGet.Tests.Apex.csproj(0,0): Error NU1100: Unable to resolve 'MSTest.TestAdapter (>= 1.2.0)' for '.NETFramework,Version=v4.7.2/x86'. PackageSourceMapping is enabled, the following source(s) were not considered: dotnet-eng, dotnet-public, myget-legacy@Local, nuget-build, vside_vs-impl, vside_vssdk.
D:\a\_work\1\s\test\NuGet.Tests.Apex\NuGet.Tests.Apex\NuGet.Tests.Apex.csproj : error NU1100: Unable to resolve 'MSTest.TestAdapter (>= 1.2.0)' for '.NETFramework,Version=v4.7.2/x86'. PackageSourceMapping is enabled, the following source(s) were not considered: dotnet-eng, dotnet-public, myget-legacy@Local, nuget-build, vside_vs-impl, vside_vssdk. [D:\a\_work\1\s\build\restorehelper.targets]
##[error]test\NuGet.Tests.Apex\NuGet.Tests.Apex\NuGet.Tests.Apex.csproj(0,0): Error NU1100: Unable to resolve 'MSTest.TestFramework (>= 1.2.0)' for '.NETFramework,Version=v4.7.2/x86'. PackageSourceMapping is enabled, the following source(s) were not considered: dotnet-eng, dotnet-public, myget-legacy@Local, nuget-build, vside_vs-impl, vside_vssdk.
D:\a\_work\1\s\test\NuGet.Tests.Apex\NuGet.Tests.Apex\NuGet.Tests.Apex.csproj : error NU1100: Unable to resolve 'MSTest.TestFramework (>= 1.2.0)' for '.NETFramework,Version=v4.7.2/x86'. PackageSourceMapping is enabled, the following source(s) were not considered: dotnet-eng, dotnet-public, myget-legacy@Local, nuget-build, vside_vs-impl, vside_vssdk. [D:\a\_work\1\s\build\restorehelper.targets]
  Failed to restore D:\a\_work\1\s\test\NuGet.Tests.Apex\NuGet.OptProf\NuGet.OptProf.csproj (in 1.84 sec).
  Failed to restore D:\a\_work\1\s\test\NuGet.Tests.Apex\NuGet.Tests.Apex\NuGet.Tests.Apex.csproj (in 2.01 sec).
```

File: build/config.props
Comment (heng-liu) [reply build/config.props:46]: Thanks! How about adding "Ideally, this is the same .NET SDK version NuGet inserts into." before CliBranchForTesting?

File: eng/pipelines/templates/Functional_Tests_On_Windows.yml
Comment (heng-liu) [reply eng/pipelines/templates/Functional_Tests_On_Windows.yml]: Thanks! Changed to use "variables parameter on a job".

File: NuGet.Config
Comment (zivkan) [reply NuGet.Config:25]: Oops, sorry. When I first reviewed this, I thought it was `bootstrap.proj`, not `nuget.config`.  Makes sense.

File: eng/pipelines/templates/Build_and_UnitTest.yml
Comment (zivkan) [reply eng/pipelines/templates/Build_and_UnitTest.yml:225]: If there's anyone who should be notified of when changes to nuget.exe cause breaks in customer build pipelines, it's us. Also, we should know how we use nuget.exe in our pipeline (other tasks that use `nuget.exe install` to download their binaries, but we don't restore our projects with nuget.exe), so we can gauge risk.

**Review by zivkan — APPROVED** [review]

**Comment by heng-liu** [comment]
Thanks @nkolev92 @dtivel @zivkan for reviewing this PR so quickly!
Since this is blocking https://github.com/NuGet/NuGet.Client/pull/4585, I'd like to merge it soon.
Pls let me know if you have any other thoughts before merging. Thanks!

**Review by nkolev92 — APPROVED** [review]

**Pull request merged**
